core/src/test/kotlin/io/forestframework/core/http/routing/MediaTypeNegotiatingIntegrationTest.kt

```diff
@@ -0,0 +1,163 @@
+
+    ProducesAcceptCase("get 200 when matching wildcard match", produces = listOf("text/*"), accept = "text/html, application/json; level=1;q=0.9, text/plain;q=0.8", expectedStatusCode = 200),
+    ProducesAcceptCase("get 406 when matching wildcard match", produces = listOf("application/json"), accept = "text/*", expectedStatusCode = 406),
+
+    ProducesAcceptCase("get 200 when matching double wildcard case 1", produces = listOf("*/*"), accept = "*/*;level=1;q=0.9", expectedStatusCode = 200),
+    ProducesAcceptCase("get 200 when matching double wildcard case 2", produces = listOf("text/*"), accept = "*/*;q=0.7", expectedStatusCode = 200),
+    ProducesAcceptCase("get 200 when matching double wildcard case 3", produces = listOf("*/*"), accept = "text/*", expectedStatusCode = 200),
+    ProducesAcceptCase("get 200 when matching double wildcard case 4", produces = listOf("*/*"), accept = "application/xhtml+xml;q=0.7, image/jxr;level=1", expectedStatusCode = 200),
+
+    ProducesAcceptCase("get 200 when matching with parameter", produces = listOf("application/json;charset=UTF-8"), accept = "application/json;q=0.3", expectedStatusCode = 200),
+    ProducesAcceptCase("get 406 when matching with parameter", produces = listOf("application/json;charset=UTF-8"), accept = "application/json;level=1;charset=UTF-16", expectedStatusCode = 406)
+)
+
+val consumesContentTypeCases = listOf(
+    ConsumesContentTypeCase("get 200 when matching single", consumes = listOf("application/json"), contentType = "application/json", expectedStatusCode = 200),
+    ConsumesContentTypeCase("get 415 when matching single", consumes = listOf("application/json"), contentType = "text/html", expectedStatusCode = 415),
+
+    ConsumesContentTypeCase("get 200 when matching wildcard", consumes = listOf("application/json", "image/jxr"), contentType = "application/*", expectedStatusCode = 200),
+    ConsumesContentTypeCase("get 415 when matching wildcard", consumes = listOf("application/json"), contentType = "text/*", expectedStatusCode = 415),
+
+    ConsumesContentTypeCase("get 200 when matching double wildcard case 1", consumes = listOf("*/*"), contentType = "application/json", expectedStatusCode = 200),
+    ConsumesContentTypeCase("get 200 when matching double wildcard case 2", consumes = listOf("*/*"), contentType = "text/*", expectedStatusCode = 200),
+
+    ConsumesContentTypeCase("get 200 when matching with parameter", consumes = listOf("application/json;charset=UTF-8"), contentType = "application/json;q=0.9", expectedStatusCode = 200),
+    ConsumesContentTypeCase("get 415 when matching with parameter", consumes = listOf("application/json;charset=UTF-8"), contentType = "application/json;charset=UTF-16", expectedStatusCode = 415)
+)
+
+val methodNotAllowedCases = listOf(
+    MethodNotAllowedCase("get 405 if method not allowed", httpMethod = HttpMethod.POST, expectedStatusCode = 405)
```

Comment:
Several more test cases for:

- httpMethod = ALL
- httpMethod = GET and POST, request http method = GET
- httpMethod = GET and POST, request http method = PATCH

Reply:
Sure!